### PR TITLE
Youtube increment scripts

### PR DESCRIPTION
**Pull request by II245**

### CODE REVIEW

File: tools.py [hunk 1]
```diff
@@ -439,6 +440,9 @@ def filter_dataset(input_path,
 		seed):
 	dataset = transcripts.load(input_path)
 
+	if cer:
+		assert cer < 1.0, 'CER should be set in number less than 1.0'
```
Comment (vadimkantorov) [inline tools.py]: Please fuse the if condition into the assert as follows: `assert not cer or 0 <= cer < 1.0, 'CER should be unset or in range [0.0, 1)'`

File: train.py [hunk 2]
```diff
@@ -683,9 +683,8 @@ def main(args):
 		epoch, iteration = checkpoint['epoch'], checkpoint['iteration']
 		if args.train_data_path == checkpoint['args']['train_data_path']:
 			sampler.load_state_dict(checkpoint['sampler_state_dict'])
-			if args.iterations_per_epoch and iteration and iteration % args.iterations_per_epoch == 0:
```
Comment (vadimkantorov) [inline train.py:686]: why is this removed? are you still using `iterations_per_epoch` at all?

File: tools.py [hunk 3]
```diff
@@ -676,6 +705,11 @@ def find_solution_for_frontend_input_output_shapes_divisibility(
 	cmd.add_argument('--output-path', '-o', default = 'data')
 	cmd.add_argument('--comment-path', '-c')
 	cmd.set_defaults(func = processcomments)
+
+	cmd = subparsers.add_parser('cleanup_transcripts_with_empty_ref')
+	cmd.add_argument('--input-path', '-i', required = True)
+	cmd.add_argument('--emptiness', default = 1)
```
Comment (vadimkantorov) [inline tools.py]: Please rename to `min_ref_length` (and maybe change the default into 2)

File: train.py (hunk 2 above)
Comment (II245) [reply train.py:686]: In current training I not used iterations_per_epoch in train. This case causes a bug in case we train from the checkpoint with the same dataset.

File: tools.py (hunk 3 above)
Comment (II245) [reply tools.py]: done

File: train.py (hunk 2 above)
Comment (II245) [reply train.py:686]: removed iterations_per_epoch

File: tools.py (hunk 1 above)
Comment (II245) [reply tools.py]: done

File: tools.py [hunk 4]
```diff
@@ -516,12 +515,12 @@ def split(
...
 
 
-def cleanup_transcripts_with_empty_ref(input_path, emptiness):
+def cleanup_transcripts_with_empty_ref(input_path, min_ref_length):
 	transcript_array = json.load(open(input_path))
 
 	filtered = []
 	for transcript in transcript_array:
-		if len(transcript.get('ref', "")) <= emptiness:
+		if len(transcript.get('ref', "")) <= min_ref_length:
```
Comment (vadimkantorov) [inline tools.py]: `min_ref_length` treatment should be inclusive as it is in other parts of the codebase. please change the comparison to ` < min_ref_length`

File: scripts/youtube_dataset_creation_v21.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+
+TRANSCRIPT_PREPROC='--split-by-parts 4 --skip-files-longer-than-hours 4 --skip-transcript-after-seconds 3600'
+TRANSCRIBE='--mono --join-transcript --batch-time-padding-multiple 1 --align --skip-processed --max-segment-duration 4.0 --transcribe-first-n-sec 3600 --text-pipelines dataset'
+SUBSET='--gap 0.05- --cer 0.0-0.4 --duration 0.5-8.0'
+CUT="--dilate 0.025 --sample-rate $SAMPLE_RATE --mono --strip-prefix data/ --add-sub-paths --strip"
+TRAIN_TEST_SPLIT='--microval-duration-in-hours 10 --val-duration-in-hours 0 --test-duration-in-hours 0'
+
+#mkdir -p $DATASET_AUDIO
+#bash datasets/youtube.sh LIST "$YOUTUBE" > $DATASET_ROOT/youtube.txt
```
Comment (MayMeta) [inline scripts/youtube_dataset_creation_v21.sh]: Pls drop this, if it's obsolete

File: scripts/youtube_dataset_creation_v21.sh [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+#bash datasets/youtube.sh LIST 'https://www.youtube.com/channel/UCb9_Bhv37NXN1m8Bmrm9x9w'
+#head -n $SAMPLE $DATASET_ROOT/youtube.txt > $DATASET_AUDIO/audio.txt # list of links like http://youtu.be/e8bkFQD3ZhA
+#bash datasets/youtube.sh RETR $DATASET_AUDIO/audio.txt $DATASET_AUDIO
+
+#python3 datasets/youtube.py -i "$DATASET_AUDIO_MASK" -o $DATASET_AUDIO.json $TRANSCRIPT_PREPROC
```
Comment (MayMeta) [inline scripts/youtube_dataset_creation_v21.sh]: Pls, uncomment this code, and add a brief description for each stage

File: scripts/youtube_dataset_creation_v21.sh [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+## if there are any empty ref if cut.json (empty refs can cause loss issue in train time) then remove it with cleanup_transcripts_with_empty_ref tool
+# python tools.py cleanup_transcripts_with_empty_ref --input-path youtube_v21/cut/cut.json
+
+python3 tools.py split -i $DATASET_CUT_JSON -o $DATASET_CUT $TRAIN_TEST_SPLIT
+
+# in case you need 100/1000 hours dataset the script creates val and test
+#python tools.py filter_dataset --input-path youtube_v21/cut/cut_train.json --output-path youtube_v21/cut/cut_train_cer_25_1000h.json --duration-in-hours 1000 --cer 25
+
+#python3 vis.py audiosample -i $DATASET_CUT/$(basename $DATASET_CUT).json -o $DATASET_CUT.json.html
```
Comment (MayMeta) [inline scripts/youtube_dataset_creation_v21.sh]: Pls remove, if we don't use this anymore

File: tools.py (hunk 4 above)
Comment (II245) [reply tools.py]: fixed

File: scripts/youtube_dataset_creation_v21.sh (hunk 7 above)
Comment (II245) [reply scripts/youtube_dataset_creation_v21.sh]: place it to another repo

File: scripts/youtube_dataset_creation_v21.sh (hunk 6 above)
Comment (II245) [reply scripts/youtube_dataset_creation_v21.sh]: place it to another repo

File: scripts/youtube_dataset_creation_v21.sh (hunk 5 above)
Comment (II245) [reply scripts/youtube_dataset_creation_v21.sh]: place it to another repo

**Comment by MayMeta** [comment]
LGTM
